core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -1038,6 +1094,10 @@ private[kafka] class Processor(val id: Int,
    * Close the selector and all open connections
    */
   private def closeAll(): Unit = {
+    while (!newConnections.isEmpty) {
+      newConnections.poll().close()
+    }
+    newConnections.clear()
```

Comment:
`clear()` is unnecessary since we would expect the loop to clear (i.e. we shouldn't have code that clears without closing).